module5/src/main/java/Main.java

```diff
@@ -0,0 +1,154 @@
+            }
+            for (Group group : faculty.getGroups()) {
+                if (group.getStudentsList().isEmpty()) {
+                    throw new EmptyStudentListException("You haven't students in " + group.getGroupName() + " group.");
+                }
+                for (Student students : group.getStudentsList()) {
+                    if (students.getSubjects().isEmpty()) {
+                        throw new EmptySubjectException("In " + students.getStudentName() + " hasn't subjects");
+                    }
+                }
+            }
+        }
+    }
+
+    private static void calculateAverageGradeForEachStudent(List<Faculty> facultyList) {
+        List<Student> studentsList = new ArrayList<Student>();
+        for (Faculty faculty : facultyList) {
+            for (Group group : faculty.getGroups()) {
+                studentsList.addAll(group.getStudentsList());
+            }
+        }
+        for (Student student : studentsList) {
+            Map<String, Integer> subjects = student.getSubjects();
+            int sum = 0;
+            for (Integer grade : subjects.values()) {
+                sum += grade;
+            }
+            float average = sum / (float) subjects.size();
+            System.out.println(student.getStudentName() + "'s average grade is" + " - " + average);
+        }
+    }
+
+    private static void averageGradeBySubjectInOneGroup(List<Faculty> facultyList, String facultyName,
+                                                        int groupName, String subjectName) {
+        float sum = 0;
+        float average;
+        int check = 0;
+        List<Student> studentList = new ArrayList<Student>();
+        for (Faculty faculty : facultyList) {
+            if (faculty.getFacultyName().equals(facultyName)) {
+                for (Group group : faculty.getGroups()) {
+                    if (group.getGroupName() == groupName) {
+                        studentList = group.getStudentsList();
+                    }
+                }
+            }
+        }
+        for (Student student : studentList) {
+            if (student.getSubjects().containsKey(subjectName)) {
+                sum += student.getSubjects().get(subjectName);
+                check++;
+            }
+        }
+        average = sum / check;
+        System.out.println("Average grade in" + " " + facultyName + " " + "faculty," + " " + groupName + " " +
+                "group, for " + subjectName + " is - " + average);
+    }
+
+    private static void averageGradeForOneSubjectInUniversity(List<Faculty> facultyList, String enteredSubject) {
+        float sum = 0;
+        int check = 0;
+        for (Faculty faculty : facultyList) {
+            for (Group group : faculty.getGroups()) {
+                for (Student students : group.getStudentsList()) {
+                    if (students.getSubjects().containsKey(enteredSubject)) {
+                        sum += students.getSubjects().get(enteredSubject);
```

Comment:
You call `student.getSubjects(`) two times.
Create variable for this before `if` block.